demos/deleteSolution/index.html

```diff
@@ -102,30 +150,171 @@ <h1>Delete a Solution</h1>
       }
     });
 
-    require(["@esri/arcgis-rest-auth", "../dist/delete-solution-main"],
-      function (auth, main) {
+    require(["@esri/arcgis-rest-auth", "../dist/delete-solution-main", "../dist/demoCommon.js"],
+      function (auth, main, demoCommon) {
         "use strict";
-        checkFcn = check;
-        goFcn = go;
+        connectFcn = getDeployedSolutions;
+        launchDeleteSolutionsFcn = launchDeleteSolutions;
+        toggleSelectAllFcn = toggleSelectAll;
+        updateSolutionSelectionFcn = updateSolutionSelection;
         document.getElementById("input").style.display = "block";
 
-        function check () {
-          main.checkDeleteSolution(
-            document.getElementById("solutionId").value,
-            new auth.UserSession({
-              username: document.getElementById("username").value,
-              password: document.getElementById("password").value,
-              portal: document.getElementById("portal").value + "/sharing/rest"
-            })
-          ).then(
+        let haveSolutions = false;
+        let authentication = null;
+        let selectAllToggle = true;
+        let numSelected = 0;
+
+        /**
+         * Gets deployed Solutions from an organization and shows them as a checklist in the div "sourceSolutionsDiv".
+         */
+        function getDeployedSolutions() {
+          demoCommon.removeItem("sourceSolutionsListDiv");
+          authentication = null;
+          haveSolutions = false;
+          numSelected = 0;
+          selectAllToggle = true;
+          toggleBtn.innerHTML = "Select all";
+          document.getElementById("output").innerHTML = "";
+
+          const portal = (document.getElementById("portal").value || "https://www.arcgis.com") + "/sharing/rest";
+          authentication = demoCommon.getRequestAuthentication(
+            document.getElementById("username").value, document.getElementById("password").value,
+            portal
+          );
+
+          main.getDeployedSolutions(authentication)
+          .then(
+            searchResponse => {
+              var listDiv = demoCommon.addItem("sourceSolutionsDiv", "DIV", "sourceSolutionsListDiv");
+              listDiv.innerHTML = demoCommon.createChecklist(
+                searchResponse.results,
+                document.getElementById("portal").value  || "https://www.arcgis.com",
+                "sourceSolutionsList",
+                false,
+                "updateSolutionSelectionFcn(this)"
+              );
+              updateDeleteBtn();
+              document.getElementById("toggleBtn").style.display = "inline-block";
+              document.getElementById("solutionSelection").style.display = "block";
+            },
+            error => {
+              var listDiv = addItem("sourceSolutionsDiv", "DIV", "sourceSolutionsListDiv");
```

Comment:
@jmhauck, there are common UI elements among the demos, and I started to share them via demoCommon. It's called that because there was some conflict with declarations of `common` for solution-common.

Reply:
my bad on the unclear comment...I was just wondering if addItem here should be calling demoCommon.addItem here as well

Reply:
Thank you; this is an error